packages/standalone-analyzer/src/assets.ts

```diff
@@ -0,0 +1,79 @@
+import path from "node:path";
+import fs from "node:fs/promises";
+import {
+  getCompressedSize,
+  normalizePath,
+  type Asset,
+} from "@codecov/bundler-plugin-core";
+import { fileURLToPath } from "node:url";
+
+// Resolve __dirname and __filename for both CommonJS and ESModules
+let fileName: string;
+let __dirname: string;
+const isCommonJS =
+  typeof module !== "undefined" && module.exports !== undefined;
```

Comment:
why do we need this check? We've set `emitCJS: true` in our rollup config so the run code should always be CJS.